### ISSUE TITLE
FLAC BPMs are written to non-standard TEMPO tag

### ISSUE THREAD

**Issue by dangmai** [exaile/exaile#682]
Exaile's [mapping](https://github.com/exaile/exaile/blob/8fb0ebb4ea108792d38e2b8ad355b27d70e1d246/xl/metadata/flac.py#L36) for FLAC files use the `tempo` tag to store bpm information, but in all the documents that I can find, it should be stored in a `BPM` tag instead. For example, [this mapping](https://picard.musicbrainz.org/docs/mappings/) from Musicbrainz, or [this mapping](http://wiki.hydrogenaud.io/index.php?title=Tag_Mapping) from HydrogenAudio. This makes it difficult to use the BPM information in other tools, for example I use [Beets](https://beets.io/) to manage smart playlists, which does not take into account the `tempo` tag at all.
### Steps to Reproduce (for bugs)

1. Enable the BPM plugin
2. Tap the BPM to a random FLAC song
3. Use any tag reading tool (e.g. `ffprobe`) to read the tag content
4. See that the BPM is stored in the `tempo` tag

### Possible Solution

IMO just changing the tag mapping only is not enough, since it will confuse users when they re-scan their collection and see their BPM information disappear. We can either do a one time migration from the `tempo` tag to the `BPM` tag, or we can offer the option in the setting for the user to choose which tag they'd rather use, and let them handle the migration outside of Exaile if they choose to do so.

### Environment

* Operating System and version: Windows 10

* Exaile Version: Exaile 4

**Comment by sjohannes** [comment]
AFAIK our goal is to follow the MusicBrainz mapping. So yes, it's currently wrong.

Looking at the code just makes me confused.

* If FLAC uses the Vorbis tag mapping, can't we merge it into ogg.py as a subclass of OggFormat (just like OggOpusFormat)?
* The `bpm` tag mapping in ogg.py is also wrong....
* That `comment -> description` mapping also looks wrong.

We'll have to do a one-time migration, but I'm not sure what the best way to do it is. Handling conflicts will be an issue as well (if the file has both tags). And then there's also the problem that non-library files won't get fixed.

**Comment by genodeftest** [comment]
How about:
1. Reading both the standard (`BPM`) and the non-standard (`tempo`) tag, with `BPM` always winnig
2. Writing only the standard (`BPM`) tag from now on.

For when the migration happens::
* As a user, I would be surprised if my files changed on disk without me knowing, I wouldn't like a "silent" migration.
* A non-silent migration would need a good UI, something triggering it (e.g. the first time you start exaile 4.0.3 or later).
* Should we trigger on every old-style file (I highly recommend not doing this as it will be too noisy to the user) or only once on the database.

**Comment by dangmai** [comment]
I agree that silent migrations shouldn't be happening, and adding a migration window for this is IMO a bit overkill. If we follow your strategy (favoring BPM over tempo tag), I don't think it's necessary to do migrations in Exaile, we can just put in the change log that this is happening and the users should do a migration using their tool of choice. Maybe we can even provide a quick migration script in the changelog - it's a one time thing so from my point of view that should be enough.

**Comment by sjohannes** [comment]
I'm not a fan of supporting wrong tags, especially since this one is our mistake. I think we should inform the user in some way when we see these wrong tags, and offer to fix them.

How about a plugin that checks the collection—​as well as files added to the tracklist—​for these wrong tags, and then asks the user if they want to fix them? If the user doesn't want to deal with it (yet) they can disable the plugin.
